### DISCUSSION TITLE
AudioWAVServer html

### DISCUSSION THREAD

**Discussion by Ince1956** [pschatzmann/arduino-audio-tools#1847] [Q&A] [answered]
Hi
Is there a way to access the html for the AudioWAVServer or AudioWAVServerEx?

**Comment by pschatzmann** [comment, marked as answer]
There is no html: the AudioWAVServer just returns data of the format audio/wav!

**Comment by Ince1956** [comment]
Thanks for your prompt response. I am using  the
streams-i2s-webserver_wav.ino
Example

**Comment by Ince1956** [comment]
Thanks gor your prompt response.
I am using the
streams-i2s-webserver_wav.ino example. I wanted to add to the web page if
that’s not possible can you give me any hints on integrating the
AsyncWebserver with Audio-tools.

Thanks

Ian

**Comment by pschatzmann** [comment]
I think you can use both in parallel: just use different ports

**Comment by Ince1956** [comment]
Hi Phil
I am trying to run your  example streams-generator-server_wav.ino I get the
following error
TinyHttp/src/Server/HttpServer.h:327:25: error: 'class WiFiClient' has no
member named 'clear'
I tried changing clear to flush, the example compiles but the webpage
doesn't load the sound.
Any help would be appreciated.

Regards
Ian

**Comment by pschatzmann** [comment]
For what ESP32 and what Arduino core ?  
The actual version is 3.1 since a couple of days....

**Comment by Ince1956** [comment]
Hi
This is my platformIO.ini
[env:dfrobot_firebeetle2_esp32e]
platform = ***@***.***
board = dfrobot_firebeetle2_esp32e
framework = arduino
lib_deps =
    https://github.com/pschatzmann/arduino-audio-tools.git@^1.0.1
    https://github.com/pschatzmann/TinyHttp.git


build_flags = -DCORE_DEBUG_LEVEL=5 -Wno-unused-variable
-Wno-unused-but-set-variable -Wno-unused-function -Wno-format-extra-args
monitor_speed = 115200
monitor_filters = esp32_exception_decoder

**Comment by pschatzmann** [comment]
That doen't say anything: you will need to look at the compile log.
But I think you need to upgrade the platform to get an actual version.

Do you get any error messages when running the sketch ?

**Comment by Ince1956** [comment]
Is this what you mean?
CONFIGURATION:
https://docs.platformio.org/page/boards/espressif32/dfrobot_firebeetle2_esp32e.html
PLATFORM: Espressif 32 (6.9.0) > DFRobot Firebeetle 2 ESP32-E
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog,
iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h,
olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20017.241212+sha.dcc1105b
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 35 compatible libraries
Scanning dependencies...
Dependency Graph
|-- audio-tools @ 1.0.1+sha.ee55bc8
|-- TinyHttp @ 1.0.0+sha.118a695

**Comment by pschatzmann** [comment]
The log used to show the corresponding Arduino core version: That this has been dropped does not really help.
I guess they must be really far behind now...

I will have another look at this tomorrow

**Comment by Ince1956** [comment]
Ok, thanks.

**Comment by pschatzmann** [comment]
I just tried streams-generator-server_wav.ino which did not produce any errors.
Please notice that this does not need TinyHttp.

I also added a new method printVersionInfo() in the AudioTools which prints the following information:
**************************************************
AudioTools: 1.0.1
Arduino: 10812
Arduino ESP Core Version: 2.0.17
IDF Version: v4.4.7-dirty
**************************************************
for 
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20017.241212+sha.dcc1105b 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5

This is year behind of what you get with Arduino!

**Comment by Ince1956** [comment]
Thanks Phil
I’ll try it out later today
Regards
Ian

**Comment by Ince1956** [comment]
Hi Phil
Sorry to bother you, I have the following sketch which displays the sound
output on an 8x8 led matrix and also sends the audio to a web server.
 However the quality of the sound is distorted when I implement the
copier.copy() in the loop. If I comment this out the quality is excellent.
Am I doing something wrong, or is there a better way to do this?

Thanks for your
Patience.

Regards
Ian

P.S. Happy New Year



#include "AudioTools.h"
#include "AudioTools/AudioLibs/AudioRealFFT.h" // or any other supported
inplementation
#include "AudioTools\AudioLibs\LEDOutput.h"

#define PIN_LEDS 23
#define LED_X 8
#define LED_Y 8
// AudioEncodedServer server(new WAVEncoder(),"ssid","password");
AudioWAVServer server("******", "*******"); // the same a above

I2SStream i2sStream;                                    // Access I2S as
stream
ConverterFillLeftAndRight<int16_t> filler(LeftIsEmpty); // fill both
channels - or change to RightIsEmpty
VolumeStream volume(i2sStream);                         // enable volume
control
AudioRealFFT fft;                                       // or any other
supported inplementation
FFTDisplay fft_dis(fft);                                // display on matrix
LEDOutput led(fft_dis);                                 // output to LED
matrix
StreamCopy copier(fft, i2sStream);                      // for led matrix

void setup()
{
  Serial.begin(115200);
  AudioToolsLogger.begin(Serial, AudioToolsLogLevel::Error);

  // start i2s input with default configuration
  Serial.println("starting I2S...");
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.i2s_format = I2S_STD_FORMAT; // if quality is bad change to
I2S_LSB_FORMAT https://github.com/pschatzmann/arduino-audio-tools/issues/23
  config.sample_rate = 22050;
  config.channels = 2;
  config.bits_per_sample = 16;
  config.pin_bck = GPIO_NUM_26;
  config.pin_data = GPIO_NUM_21;
  config.pin_ws = GPIO_NUM_22;
  i2sStream.begin(config);
  Serial.println("I2S started");

  // Setup FFT
  auto tcfg = fft.defaultConfig();
  tcfg.length = 1024;
  tcfg.copyFrom(config);
  fft.begin(tcfg);

  fft_dis.fft_group_bin = 3;
  fft_dis.fft_start_bin = 0;
  fft_dis.fft_max_magnitude = 0x7fff;
  fft_dis.begin();

  // Setup LED matrix output
  auto lcfg = led.defaultConfig();
  lcfg.x = LED_X;
  lcfg.y = LED_Y;
  led.begin(lcfg);

  // add LEDs
  FastLED.addLeds<WS2812B, PIN_LEDS, GRB>(led.ledData(), led.ledCount());

  // set initial volume
  auto vcfg = volume.defaultConfig();
  vcfg.copyFrom(config);
  vcfg.allow_boost = true; // activate amplification using linear control
  volume.begin(vcfg);      // we need to provide the bits_per_sample and
channels
  volume.setVolume(10.0);

  // start data sink
  server.begin(volume, config, &filler);
  FastLED.setBrightness(25);
}

// Arduino loop
void loop()
{
  // Handle new connections
  server.copy();
  copier.copy();
  led.update();
}
